### PR TITLE
refactor: move CustomAppContext and CustomServerPluginHooks from shuv…

### PR DESCRIPTION
**Pull request by andy-yuanchang**
…i/runtime to shuvi/service

**What kind of change does this PR introduce?**
refactoring

**Did you add tests(unit,e2e) for your changes?**
No

**Does this PR introduce a breaking change?**
No

**What needs to be documented once your changes are merged?**
No

### CODE REVIEW

File: packages/platform-web/shuvi-type-extensions-runtime.d.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /// <reference types="@shuvi/platform-shared/shuvi-type-extensions-runtime" />
+/// <reference types="@shuvi/platform-web/shuvi-type-extensions-node" />
```
Comment (andy-yuanchang) [inline packages/platform-web/shuvi-type-extensions-runtime.d.ts:2]: Make server.ts intellisense work

File: packages/service/src/core/apiTypes.ts [hunk 2]
```diff
@@ -115,6 +115,9 @@ export type IPlatform = (
 export interface CustomCorePluginHooks
   extends ShuviService.CustomCorePluginHooks {}
 
+export interface CustomServerPluginHooks
```
Comment (liximomo) [inline packages/service/src/core/apiTypes.ts:118]: 这个声明应该直接放到  service/src/server/pluginTypes.ts 中

**Review by liximomo — CHANGES_REQUESTED** [review]

File: packages/platform-web/src/node/features/html-render/shuvi-app.ts [hunk 3]
```diff
@@ -1,14 +1,7 @@
-import { extendedHooks } from './serverHooks';
-
+import '@shuvi/runtime';
```
Comment (liximomo) [inline packages/platform-web/src/node/features/html-render/shuvi-app.ts:1]: 这行感觉是多余的？ 你试试改成 export {}

**Review by liximomo — CHANGES_REQUESTED** [review]

**Review by liximomo — CHANGES_REQUESTED** [review]
/lgtm

**Review by liximomo — APPROVED** [review]

**Pull request merged**